src/expression_evaluator/function_evaluator.cpp

```diff
@@ -21,8 +21,16 @@ void FunctionExpressionEvaluator::evaluate() {
     for (auto& child : children) {
         child->evaluate();
     }
+    auto expr = reinterpret_cast<binder::ScalarFunctionExpression*>(expression.get());
+    if (expr->getFunctionName() == CAST_FUNC_NAME &&
+        parameters[0]->dataType.getLogicalTypeID() == LogicalTypeID::STRING) {
+        execFunc(parameters, *resultVector,
+            &reinterpret_cast<function::StringCastFunctionBindData*>(expr->getBindData())
+                 ->csvConfig);
+        return;
+    }
     if (execFunc != nullptr) {
-        execFunc(parameters, *resultVector);
+        execFunc(parameters, *resultVector, nullptr);
```

Comment:
You can make the  dataptr default type to null, so you don't need to pass in the nullptr there.

Reply:
Since 
`using scalar_exec_func = std::function<void(
    const std::vector<std::shared_ptr<common::ValueVector>>&, common::ValueVector&, void*)>;`
specifies the function needs to take in 3 arguments, I still need to pass `nullptr`.